gnrc: cleanup of gnrc_pkt and enhance unittest

smlng opened the pull request:
minor optimization and correction, e.g. `/ptr/pkt/` like in header file.

kaspar030 commented:
Please describe your optimization a little in the PR description.

miri64 commented on sys/net/gnrc/pkt/gnrc_pkt.c:
```diff
@@ -11,22 +12,18 @@
-            return ptr;
-        }
-        ptr = ptr->next;
+    while (pkt && (pkt->type != type)) {
```
I would still prefer `(pkt != NULL)` instead of just `pkt`, to prevent type-mixing.

smlng commented:
@kaspar030 and @miri64 please (re)review, I fixed the title and first commit message, its more a minor cleanup and corrects, i.e., remove unused `assert.h`, rename `/ptr/pkt/` for consistency, and make pointer checks for not-NULL explicit. 

Further, I found that the corresponding unittest was missing a test of the search function, so I added that, too.

miri64 commented on tests/unittests/tests-pkt/tests-pkt.c:
```diff
@@ -106,6 +109,25 @@ static void test_pkt_count__null(void)
     TEST_ASSERT_EQUAL_INT(0, gnrc_pkt_count(NULL));
 }
 
+static void test_pkt_type(void)
```
Please name the test like the function you test is called:

```
test_<function name>[__subhint]()
```

miri64 commented on tests/unittests/tests-pkt/tests-pkt.c:
```diff
@@ -106,6 +109,25 @@ static void test_pkt_count__null(void)
+    gnrc_pktsnip_t snip3 = _INIT_ELEM_STATIC_TYPE(GNRC_NETTYPE_IPV6, &snip2);
+    /* successfull searches */
+    TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_UNDEF,
+                          gnrc_pktsnip_search_type(&snip3, GNRC_NETTYPE_UNDEF)->type);
```
Maybe check, if the function returns a non-NULL value to prevent (for the tester) unclear crashes in an error case.

smlng replied:
you mean something like:
```
TEST_ASSERT(gnrc_pktsnip_search_type(&snip3, GNRC_NETTYPE_UNDEF) != NULL);
TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_UNDEF,
                          gnrc_pktsnip_search_type(&snip3, GNRC_NETTYPE_UNDEF)->type);
```

miri64 replied:
Maybe more like:
```
gnrc_pktsnip_t *res;

TEST_ASSERT_NOT_NULL((res == gnrc_pktsnip_search_type(&snip3, GNRC_NETTYPE_UNDEF)));
TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_UNDEF, res->type);
```

miri64 replied:
(Oops, `=` instead of `==` I meant [edited reply accordingly])

smlng replied:
will do!

smlng commented:
@miri64 addressed your comments.
